sources/packages/web/src/components/common/IdleTimeChecker.vue

```diff
@@ -6,19 +6,33 @@
     @keyup="setLastActivityTime"
   >
     <slot></slot>
-    <ConfirmExtendTime ref="extendTimeModal" :clientIdType="clientIdType" />
+    <ConfirmExtendTime
+      ref="extendTimeModal"
+      :clientIdType="clientIdType"
+      :countdown="countdown"
+      :showDialog="modalOpen"
+      @dialogClosedEvent="extendUserSessionTime"
+    />
   </div>
 </template>
 
 <script lang="ts">
-import { onMounted, ref, computed } from "vue";
+import { onMounted, ref, computed, watch } from "vue";
 import { ClientIdType } from "@/types";
-import { ModalDialog, useInstitutionAuth, useFormatters } from "@/composables";
 import {
-  MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER,
-  MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT,
-  MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION,
-  MINIMUM_IDLE_TIME_FOR_WARNING_AEST,
+  ModalDialog,
+  useInstitutionAuth,
+  useAuth,
+  useFormatters,
+} from "@/composables";
+import {
+  MAXIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER,
+  MAXIMUM_IDLE_TIME_FOR_WARNING_STUDENT,
+  MAXIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION,
+  MAXIMUM_IDLE_TIME_FOR_WARNING_AEST,
+  COUNT_DOWN_TIMER_FOR_LOGOUT,
+  LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM,
+  LOGGED_OUT_LOCAL_STORAGE_ITEM,
```

Comment:
If these constants are used only in this scope they can be declared only in this file.
The exception to this is usually error constants that will be mostly of the time declared to be shared.
I would recommend declaring these ones in the file scope only.
To be clear, I am talking about `LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM` and `LOGGED_OUT_LOCAL_STORAGE_ITEM`

Reply:
I think we should use LOGGED_OUT_LOCAL_STORAGE_ITEM for when the user clicks on the logout in the main menu. Then other tabs would detect it and log themselves out too.

Reply:
That did not happen at least in my local test. The second tab was still logged.

Reply:
You probably clicked on the logout in the main menu. That's what I mean. I didn't change that but I would ask if we should do that for the main logout option too (I think we should). 😄

Reply:
That is a nice catch 😉